metadata_layer.py

```diff
@@ -456,7 +455,18 @@ def mirror_link(self, model_object, data_source, link, link_obj, policy):
         # If we fetched the representation and it hasn't changed,
         # the previously mirrored version is fine. Don't mirror it
         # again.
-        if representation.status_code == 304:
+        if representation.status_code == 304 and representation.mirror_url:
+            return
+
+        # The metadata may have some idea about the media type for this
+        # LinkObject, but the media type we actually just saw takes 
+        # precedence.
+        if representation.media_type:
+            link.media_type = representation.media_type
+
```

Comment:
I moved this up so that we store the media type in the database even if the representation turns out not to be mirrorable.